app/src/main/java/de/rampro/activitydiary/db/ActivityDiaryContentProvider.java

```diff
@@ -434,4 +441,77 @@ public int update(@NonNull Uri uri, @NonNull ContentValues values, @Nullable Str
     public void resetDatabase() {
         mOpenHelper.close();
     }
+
+
+    /**
+     Search for all dates in database which match start/end date or are in range (between start and end date)
+     * @param date - date is searched
+     * @param dateFormat - format under we search and compare matches
+     * @return query (string) with ids that fulfills defined conditions
+     */
+    public String searchDate(String date, String dateFormat) {
+        String querySelection = " ";
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat);
+        Calendar searchedValueCal = Calendar.getInstance();
+        Calendar startValueCal = Calendar.getInstance();
+        Calendar endValueCal = Calendar.getInstance();
+
+        //Gets all start, end dates and also id of activities
+        try {
+            Cursor allRowsStart =  mOpenHelper.getReadableDatabase().rawQuery(
```

Comment:
a raw query to get ALL activities in the diary seems to be too much... Please convert the date into a timestamp and search for that, this  will dramatically increase the performance, because the database can use things like an index and compare numbers as native code instead of creating thousands of java objects.

Reply:
ok, you let me know if you are stuck..

Reply:
START and END are stored as INT (64 bit), which represent milliseconds since epoch (what you get with e. g. System.currentTimeMillis()) or Date.getTime()

Reply:
or something liek new GregorianCalendar(2013,10,31).getTimeInMillis() will give you the beginning of the 31th of October 2013 in the "timestamp"-format used for START or END...